### ISSUE TITLE
Track Postgres EOL lifecycle

### ISSUE THREAD

**Issue by jqnatividad** [ckan/ckan#5854]
**CKAN version**
2.9.1

**Describe the bug**
The current minimum postgres version for CKAN is 9.2.  Postgres 9.2 is no longer supported.

**Expected behavior**
The current release should ideally always specify a supported Postgres version.  Similar to how CKAN tracks the Ubuntu LTS version, perhaps, there should also be a CKAN tracking policy consistent with Postgres' EOL lifecycle.

https://endoflife.software/applications/databases/postgresql

**Comment by jqnatividad** [comment]
Hi @kowh-ai , 
Mea culpa.  I missed the doc reference. 

But I did find a version check for at least 9.2 in the datastore backend -  https://github.com/ckan/ckan/blob/59726c8c38ea1c08cf2d69fbcf7c7571cbd17d1a/ckanext/datastore/backend/postgres.py#L299-L306

Also, 9.5 is being EOLed Feb 11, 2021, with 9.6 EOLing shortly after in Nov 11, 2021.  Is the team considering requiring v10 for the next major version release (v3.x +)?

**Comment by Zharktas** [comment]
I think we should follow whatever ubuntu has as long as our install docs are based on ubuntu. 16.04 (which EOLs in april) has 9.5, 18.04 has 10 and 20.04 has 12.

**Comment by kowh-ai** [comment]
Yes @Zharktas good idea - I could update docs to use whatever version is in Ubuntu's default repository and mention about staying in support using @jqnatividad 's link: https://endoflife.software/applications/databases/postgresql

**Comment by jqnatividad** [comment]
Great @kowh-ai!  While you're at it, what about creating an explicit EOL lifecycle section in the documentation?

Explicitly stating what @Zharktas mentioned, as well as how long older versions of CKAN will be supported, with deadlines?

And even though CKAN doesn't do [cadence releases like Ubuntu](https://ubuntu.com/about/release-cycle), maybe consider preemptively adding at least the next release in this lifecycle section for planning purposes.

**Comment by jqnatividad** [comment]
Going back to the hardcoded 9.2 check in the datastore, should I create a PR for that?

**Comment by kowh-ai** [comment]
@jqnatividad - please, that would be great!

**Comment by jqnatividad** [comment]
Actually, as the datastore could potentially be a different db server than the one CKAN uses, and the version check above is more about postgres 9.2 introducing support for nested json, I decided not to submit a PR.

Still, this was a useful discussion as it will result in having explicit EOL guidance in the docs - keeping the ticket open until then.

**Comment by jqnatividad** [comment]
@kowh-ai ended doing a min postgres version check after all in the Datastore.

Planning ahead, as @Zharktas pointed out - Ubuntu 16.04 is EOLing in April and 18.04 bundles postgres v10, the team should consider making postgres v10.15 the minimum required version.

**Comment by duttonw** [comment]
What's the Postgres supported version range that 2.10, 2.11 should now be supporting?
